### ISSUE TITLE
Add 'none' update reconciliation strategy

### ISSUE THREAD

**Issue by dippynark** [fluxcd/helm-controller#268]
Originally raised on Slack: https://cloud-native.slack.com/archives/CLAJ40HV3/p1622138440000400

## What happened?

We are using the helm-controller’s HelmRelease resource (v2beta1) and the Nginx Ingress Controller. All the Pods of the Nginx Ingress Controller were down and we upgraded a chart that included an Ingress resource; the upgrade failed due to the call the Nginx Ingress Controller's webhook failing and then rollback remediation failed for the same reason. This meant that the HelmRelease was stuck even though we have `retries: -1` for upgrade remediation. Looking at the code it seems that if remediation fails once then the helm-controller stops forever (https://github.com/fluxcd/helm-controller/blob/577925cf8384810916f2b4d285f546a09429b125/controllers/helmrelease_controller.go#L320).

This same situation could happen for any webhook that becomes temporarily unavailable.

## Solution

I understand that in some cases retrying can be unsafe, but for most cases in my opinion it’s better to keep trying instead of requiring manual intervention (after all, the chart in question had no issues, it was just an unfortunately timed upgrade).
We thought about changing remediation strategy to `uninstall`, but this is not ideal as it would cause downtime for our chart (and would also delete CRDs, potentially causing more problems).

Ideally, we could set no remediation strategy and just attempt the upgrade indefinitely (this feels more closely aligned to the GitOps methodology), so I think a good solution would be implement a `none` upgrade reconciliation strategy which always succeeds, allowing update retries to continue.

## Alternatives

Create a CronJob that runs `helm rollback`  on HelmReleases that have failed their rollback. Of course this is not ideal and remediation logic should stay within the controller.

**Comment by hiddeco** [comment]
If I am following you correctly you want to retry indefinitely without remediating? What would you think of `Retry` as a strategy name?

If the upgrade fails, do you expect it to requeue on the defined interval, or a retry exponential backoff, something else?

> It seems this was possible using the helm-operator, so maybe there was a reason for removing this feature: docs.fluxcd.io/projects/helm-operator/en/stable/helmrelease-guide/rollbacks/#enabling-retries-of-rolled-back-releases

If no retries for rolled back releases are enabled, they require human intervention as well to continue the release process.

**Comment by dippynark** [comment]
Yeah, as is standard with a controller it should be able to drive towards the desired state despite transient failures. The currently implementation is very brittle.

`Retry` doesn't make sense to me as the subsequent upgrade is the retry, not the lack of remediation.

Re-queue and with exponential backoff yes. The interval field should be removed imo (everything the helm-controller needs to react to is already a Kubernetes event).
